serviceaccounts.rb

```diff
@@ -35,33 +39,29 @@ def existing_file_account(path)
   def run()
     common = Common.new
     ENV["GOOGLE_APPLICATION_CREDENTIALS"] = @path
-    service_account = @service_account
-    if not service_account
-      service_account = "#{@project}@appspot.gserviceaccount.com"
-    end
-    if service_account == existing_file_account(@path)
+    if @service_account == existing_file_account(@path)
       # Don't generate another key if this account is already active. This can
       # happen for nested service account contexts, for example.
       common.status "Attaching to existing keyfile @ #{@path}"
       yield
       return
     end
 
-    if service_account == "all-of-us-workbench-test@appspot.gserviceaccount.com"
+    if @service_account == "all-of-us-workbench-test@appspot.gserviceaccount.com"
       unless File.exists?(@path)
         common.run_inline %W{gsutil cp gs://#{@project}-credentials/app-engine-default-sa.json
```

Comment:
nit: I bet there's a more pure Ruby way to do this with a client lib.

Reply:
I'm sure there is. Ruby's primary role in the codebase is to orchestrate calls to other command lines, so I don't think that invoking gcloud or gsutil over a client lib is incorrect. Currently it also requires 0 gem dependency installs, it would be nice to keep it lightweight.

Reply:
Sorry, missed your last round of comments until just now.

There's definitely a conversation to be had around how and how much we use Ruby. It doesn't sound like it's anyone's favorite choice currently. For my part, if I'm going to use it, I want to use it to the fullest, and using actual client libs instead of curl and other system calls seems like an important part of that. If it's just a better bash, then I guess that works too, but if there's a way to work at a higher level of abstraction, that's always nice.

FWIW I would trade pulling in gems from google over git module imports.